### PR TITLE
Implement pyopencl.array.stack

### PR DESCRIPTION
**Pull request by kaushikcfd**

### CODE REVIEW

File: pyopencl/array.py [hunk 1]
```diff
@@ -2579,6 +2579,38 @@ def hstack(arrays, queue=None):
...
     return result
 
+
+def stack(arrays, axis=0, queue=None):
+    if not arrays:
+        raise ValueError("need at least one array to stack")
+
+    if axis != 0:
+        # pyopencl.Array.__setitem__ does not support non-contiguous assignments
+        raise NotImplementedError("axis!=0 not implemented")
```
Comment (inducer) [inline pyopencl/array.py]: This is only true for C-layout arrays. (`ary.flags.c_contiguous`) So IMO this needs to be checked.

Comment (inducer) [inline pyopencl/array.py]: You can easily generalize this to Fortran-layout arrays `ary.flags.f_contiguous` by transposing inputs and outputs. (should test this as well)

File: pyopencl/array.py [hunk 2]
```diff
@@ -2579,6 +2579,38 @@ def hstack(arrays, queue=None):
 
     return result
 
+
+def stack(arrays, axis=0, queue=None):
```
Comment (inducer) [inline pyopencl/array.py:2583]: Add to docs.

**Review by inducer — COMMENTED** [review]
Thanks! Just a few nits to pick, good once those are done.

**Comment by inducer** [comment]
Unsubscribing... @-mention or request review once it's ready for a look or needs attention.

File: pyopencl/array.py (hunk 2 above)
Comment (kaushikcfd) [reply pyopencl/array.py:2583]: f7c6bff226a55c1e4b95cf71bd819ffb9fe1c8f9

File: pyopencl/array.py (hunk 1 above)
Comment (kaushikcfd) [reply pyopencl/array.py]: Yep, works for f_contiguous arrays too: ad6b6957a0fef0e65b55660311d5830a2c72e8b5

Comment (kaushikcfd) [reply pyopencl/array.py]: Yep, wasn't correct for all array layout orderings. Fixed in ad6b6957a0fef0e65b55660311d5830a2c72e8b5.

**Comment by kaushikcfd** [comment]
@inducer: Thanks, this is ready for another review!

**Comment by inducer** [comment]
Thx!

**Pull request merged**